File: practica1/Ejercicio3/src/ejercicio3/Empleado.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ejercicio3;

/**
 *
 * @author carme
 */
public class Empleado {
    String nombre;
    String dni;
    String cargo;
    String tipoContrato;


    
    Empleado(String nombre, String dni, String cargo, String tipoContrato)
    {
        this.nombre = nombre;
        this.dni = dni;
        this.cargo = cargo;
        this.tipoContrato = tipoContrato;
    }
    
    public String toString(){
        String string="";
        string += "Nombre: "+ this.nombre+" \n";
        string += "\tDNI: "+ this.dni+" \n";
        string += "\tCargo: "+ this.cargo+" \n";
        string += "\tContrato actual: "+ this.tipoContrato+" \n";
        return string;
    };
    
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setDni(String dni) {
        this.dni = dni;
    }

    public void setCargo(String cargo) {
        this.cargo = cargo;
    }

    public void setTipoContrato(String tipoContrato) {
        this.tipoContrato = tipoContrato;
    }

    // Si vemos que no hace falta esto se quita
    public void addEmpleado(Empleado empleado){
        System.out.println("Un empleado no puede contener elementos jerarquicos");
    };
    public void addDepartamento(Departamento departamento){
        System.out.println("Un empleado no puede contener elementos jerarquicos");
    };
    public void removeEmpleado(String dni){
        System.out.println("Un empleado no puede contener elementos jerarquicos");
    };
    public void removeDepartamento(String nombre){
        System.out.println("Un empleado no puede contener elementos jerarquicos");
    };
   
}

File: practica2/ejercicio_grupal/lib/empleado.dart
import 'elementoEmpresa.dart';

class Empleado extends ElementoEmpresa {
  String nombre = '';
  String dni = '';
  String cargo = '';
  String tipoContrato = '';

  Empleado(nombre, String dni, String cargo, String tipoContrato)
  {
    this.nombre = nombre;
    this.dni = dni;
    this.cargo = cargo;
    this.tipoContrato = tipoContrato;
  }

  @override
  String mostrarJerarquia(){
    String string="Empleado:\n";
    string += "\tNombre: "+ this.nombre+" \n";
    string += "\tDNI: "+ this.dni+" \n";
    string += "\tCargo: "+ this.cargo+" \n";
    string += "\tContrato actual: "+ this.tipoContrato+" \n";
    return string;
  }

  void setNombre(String nombre) {
    this.nombre = nombre;
  }

  void setDni(String dni) {
    this.dni = dni;
  }

  void setCargo(String cargo) {
    this.cargo = cargo;
  }

  void setTipoContrato(String tipoContrato) {
    this.tipoContrato = tipoContrato;
  }

  @override
  void addElementoEmpresa(ElementoEmpresa elemento)
  {

  }

  @override
  void removeElementoEmpresa(ElementoEmpresa elemento)
  {

  }

  @override
  bool? esSuPadre(ElementoEmpresa departamento)
  {

  }

  @override
  void cambiarSuperior(ElementoEmpresa nuevoSuperior)
  {

  }

  @override
  ElementoEmpresa getElementoEmpresa(int index)
  {
    return this;
  }

  @override
  String toString(){
    String s = this.nombre;
    return s;
  }

}